### ISSUE TITLE
Sporadic ~1s freezes(video only) when using hwdec=vaapi

### ISSUE THREAD

**Issue by yeahnope** [mpv-player/mpv#13079]
### Important Information

Provide following Information:
- mpv version
  - mpv 0.37.0 (via apt), mpv v0.37.0-86-g8f22916f85 (built via mpv-build)
- Linux Distribution and Version
  - Debian trixie Kernel 6.5.0-5-amd64
- Source of the mpv binary
  - problem happens with the apt packaged version as well as the selfbuilt version
- If known which version of mpv introduced the problem
  - unknown
- Window Manager and version
  - sway version 1.8.1
- GPU driver and version
  - amdgpu / mesa 23.2.1
- Possible screenshot or video of visual glitches
  - N/A

If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps

`mpv --no-config --hwdec=vaapi`

### Expected behavior

Video runs without dropped frames/without freezes

### Actual behavior

Video runs fine most of the time but sporadically (approximately once in 5-10 minutes) freezes for about 1 or 2 seconds. mpv logs dont show anything about it though and mpv lists 0 dropped frames. Also it seems like the frame shown during the freeze is not the most recent one but rather a buffer from a few frames prior. Audio is not interrupted.

### Log file

[freeze.txt](https://github.com/mpv-player/mpv/files/13647785/freeze.txt)

### Sample files

This happened with any vp9 video that was long enough, the example used in the log file is https://www.youtube.com/watch?v=hhAtN_rRuQo

**Comment by yeahnope** [comment]
This also happens on weston 12.0.1, so it does not seem to be a sway issue

**Comment by llyyr** [comment]
Check `dmesg` to see if there was a gpu reset when you get these freezes

**Comment by yeahnope** [comment]
> Check `dmesg` to see if there was a gpu reset when you get these freezes

I did and theres nothing in dmesg when this happens. Also I've realized that the gpu firmware I was using wasn't the most recent one so I manually replaced the files I was using with the ones from [the firmware repo](https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/amdgpu) but sadly the problem still persists.

However I have not been able to reproduce the problem with `vo=dmabuf-wayland` (and `hwdec=vaapi`), which has a few different issues for me but maybe someone knows whats different between `vo=gpu` and `vo=dmabuf-wayland` that might produce such freezes?
